subsys/mgmt/mcumgr/Kconfig

```diff
@@ -101,6 +101,30 @@ config MGMT_MAX_DECODING_LEVELS
 	  size of cbor_nb_reader structure by zcbor_state_t size per
 	  one unit selected here.
 
+config MGMT_MCUMGR_MGMT_NOTIFICATION_HOOKS
+	bool "MCUMGR notification hook support"
+	help
+	  With this enabled, applications and parts of code can register for MCUMGR event
+	  notifications which will result in callbacks when a registered event occurs. Note that
+	  this enables the base notification functionality but itself does not enable any
+	  notifications, which must be enabled by selecting other Kconfig options.
+
+	  To enable notifications in code, mgmt_register_callback() must be called with the
+	  callback function and events that want to be received. Multiple handlers can be
+	  registered and will all be called when registered events occur.
+
+	  Some callbacks support notifying the calling function of a status, in which to accept
+	  or decline the current operation, by returning false this will signal to the calling
+	  function that the request should be denied, for informal-only notifications or
+	  acceptable, true must be returned by all the registered notification handlers.
+
+config MGMT_MCUMGR_SMP_COMMAND_STATUS_HOOKS
```

Comment:
Sorry for asking to change this again, but I have agreed that the prefix `MGMT_` is too much, and `MCUMGR_` is distinctive enough.